### PR TITLE
show diffs for stash items

### PR DESCRIPTION
**Pull request by campersau**
reference #444

Clicking on the stash title will show the changed files.

image

image

### CODE REVIEW

**Comment by campersau** [comment]
@codingtwinky wouldn't it be better to move the `Default` and `Side by Side` buttons and the `textDiffType` logic to the `commitdiff` component?

Because these things are now duplicated in the `stash` and `commit` component.

**Comment by FredrikNoren** [comment]
Really nice! Only two comments from me are the usual suspects: add to CHANGELOG and, if we can, a basic click-test so it doesn't break in the future

**Comment by codingtwinky** [comment]
@campersau I do like the thought and if we do that we can reduce duplicate logic.  we can also further remove same duplicate logic via recycling comittdiff component and use it in staging, right now there are many duplicate logics between diff for staging and diff for commits as code base is different. 

If you would like to do that with in this PR please let me know so I can pause #725 as i'd imagine there will be many conflicts.

**Comment by campersau** [comment]
@codingtwinky okay lets do this after this PR and #725 are merged.

**Comment by codingtwinky** [comment]
@campersau #725 is merged, have you had the chance to work on this yet?  Is there anything I can do to help?  I think this would be nice...

**Comment by campersau** [comment]
@codingtwinky I will try to work on this this week.

File: components/staging/staging.html [hunk 1]
```diff
@@ -49,20 +49,20 @@
...
-            <button class="btn btn-default bootstrap-tooltip" data-ta-clickable="stage-wordwrap" data-bind="click: toggleWordWrap.bind($data, true), css: {active: wordWrap}" data-toggle="tooltip" data-placement="bottom" data-original-title="Wrap words per line" data-delay='{"show":"2000", "hide":"0"}'>
-              <span data-bind="text: 'Word Wrap'"></span>
-            </button>
             <button class="btn btn-default bootstrap-tooltip" data-ta-clickable="stage-nowrap" data-bind="click: toggleWordWrap.bind($data, false), css: {active: !wordWrap()}" data-toggle="tooltip" data-placement="bottom" data-original-title="Not wrapping words per line" data-delay='{"show":"2000", "hide":"0"}'>
               <span data-bind="text: 'Default'"></span>
             </button>
+            <button class="btn btn-default bootstrap-tooltip" data-ta-clickable="stage-wordwrap" data-bind="click: toggleWordWrap.bind($data, true), css: {active: wordWrap}" data-toggle="tooltip" data-placement="bottom" data-original-title="Wrap words per line" data-delay='{"show":"2000", "hide":"0"}'>
+              <span data-bind="text: 'Word Wrap'"></span>
+            </button>
           </div>
```
Comment (campersau) [inline components/staging/staging.html]: I just moved these things around to be consistent with the other buttons. (Show `Default` as first option.)

**Comment by campersau** [comment]
I think this is good to go now:
- moved the `Default` and `Side by Side` buttons and the `textDiffType` logic to the `commitdiff` component
- added clicktests for the `stash` component. stash -> open diff -> pop stash

**Comment by codingtwinky** [comment]
:+1:

Thanks for moving the toggle button logic to commit diff!

**Pull request merged**